tests/integration/integration_test.py

```diff
@@ -1173,100 +1162,69 @@ def verify_config(expconfig, configs):
 
 
 def verify_avro_explicit_read_schema():
-    """ verify that reading Avro with explicit reader schema works"""
     from confluent_kafka import avro
-    avsc_dir = os.path.join(os.path.dirname(__file__), os.pardir, 'avro')
 
-    # Producer config
-    conf = {'bootstrap.servers': bootstrap_servers,
-            'error_cb': error_cb}
+    """ verify that reading Avro with explicit reader schema works"""
+    base_conf = {'bootstrap.servers': bootstrap_servers,
+                 'error_cb': error_cb,
+                 'api.version.fallback.ms': 0,
+                 'broker.version.fallback': '0.11.0.0',
+                 'schema.registry.url': schema_registry_url}
 
-    # Create producer
-    if schema_registry_url:
-        conf['schema.registry.url'] = schema_registry_url
-        p = avro.AvroProducer(conf)
-
-    key_schema = avro.load(os.path.join(avsc_dir, "primitive_float.avsc"))
-    schema1 = avro.load(os.path.join(avsc_dir, "user_v1.avsc"))
-    schema2 = avro.load(os.path.join(avsc_dir, "user_v2.avsc"))
-    float_value = 32.
-    val = {
-        "name": "abc",
-        "favorite_number": 42,
-        "favorite_colo": "orange"
+    consumer_conf = dict(base_conf, **{
+        'group.id': 'test.py',
+        'session.timeout.ms': 6000,
+        'enable.auto.commit': False,
+        'on_commit': print_commit_result,
+        'auto.offset.reset': 'earliest',
+        'schema.registry.url': schema_registry_url})
+
+    avsc_dir = os.path.join(os.path.dirname(__file__), os.pardir, 'avro')
+    writer_schema = avro.load(os.path.join(avsc_dir, "user_v1.avsc"))
+    reader_schema = avro.load(os.path.join(avsc_dir, "user_v2.avsc"))
+
+    user_value1 = {
+        "name": " Rogers Nelson"
     }
-    val1 = {
-        "name": "abc"
+
+    user_value2 = {
+        "name": "Kenny Loggins"
     }
 
     combinations = [
-        dict(value=val, value_schema=schema2, key=float_value, key_schema=key_schema,
-             reader_value_schema=schema1, reader_key_schema=key_schema),
-        dict(value=val1, value_schema=schema1, key=float_value, key_schema=key_schema,
-             reader_value_schema=schema2, reader_key_schema=key_schema),
+        dict(key=user_value1, key_schema=writer_schema, value=user_value2, value_schema=writer_schema),
+        dict(key=user_value2, key_schema=writer_schema, value=user_value1, value_schema=writer_schema)
     ]
+    topic = str(uuid.uuid4())
```

Comment:
use the testconf topic prefix to be able to identify topics created by these tests